applications/MeshingApplication/custom_processes/multiscale_refining_process.cpp

```diff
@@ -0,0 +1,848 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "custom_processes/multiscale_refining_process.h"
+#include "geometries/point.h"
+#include "processes/fast_transfer_between_model_parts_process.h"
+#include "utilities/assign_unique_model_part_collection_tag_utility.h"
+#include "custom_utilities/meshing_flags.h"
+
+namespace Kratos
+{
+
+MultiscaleRefiningProcess::MultiscaleRefiningProcess(
+    ModelPart& rThisCoarseModelPart,
+    ModelPart& rThisRefinedModelPart,
+    ModelPart& rThisVisualizationModelPart,
+    Parameters ThisParameters)
+    : mrCoarseModelPart(rThisCoarseModelPart)
+    , mrRefinedModelPart(rThisRefinedModelPart)
+    , mrVisualizationModelPart(rThisVisualizationModelPart)
+    , mParameters(ThisParameters)
+    , mUniformRefinement(mrRefinedModelPart)
+{
+    Parameters DefaultParameters = Parameters(R"(
+    {
+        "number_of_divisions_at_subscale"     : 2,
+        "echo_level"                          : 0,
+        "subscale_interface_base_name"        : "refined_interface",
+        "subscale_boundary_condition"         : "Condition2D2N"
+    }
+    )");
+
+    mParameters.ValidateAndAssignDefaults(DefaultParameters);
+
+    mDivisionsAtSubscale = mParameters["number_of_divisions_at_subscale"].GetInt();
+    mEchoLevel = mParameters["echo_level"].GetInt();
+
+    std::string interface_base_name = mParameters["subscale_interface_base_name"].GetString();
+    mRefinedInterfaceName = interface_base_name + "_" + std::to_string(mrCoarseModelPart.GetValue(SUBSCALE_INDEX) + 1);
+    mInterfaceConditionName = mParameters["subscale_boundary_condition"].GetString();
+
+    if (mEchoLevel > 1) KRATOS_WATCH(mParameters);
```

Comment:
```suggestion
    KRATOS_INFO_IF("MultiscaleRefiningProcess", mEchoLevel > 1) << "Parameters\n" << mParameters.Info() << std::endl;
```

Reply:
Now it is possible on GitHub to apply directly suggested changes, so you don't need to change the code in your machine and apply suggestion from other people directly